gstd/src/common/handlers.rs

```diff
@@ -25,50 +25,209 @@
-            ),
-            (Some(msg), None) => format!("'{msg:?}'"),
-            (None, Some(loc)) => {
-                format!("{}:{}:{}", loc.file(), loc.line(), loc.column())
+        #[panic_handler]
+        pub fn panic(panic_info: &PanicInfo) -> ! {
+            use crate::prelude::{fmt::Write, mem::MaybeUninit, str};
+            use arrayvec::ArrayString;
+
+            let option = panic_info.message().zip(panic_info.location());
+            let (message, location) = unsafe { option.unwrap_unchecked() };
+
+            fn itoa_u32(buffer: &mut [MaybeUninit<u8>; 10], mut n: u32) -> &str {
+                let mut idx = buffer.len();
+                loop {
+                    idx -= 1;
+                    unsafe { buffer.get_unchecked_mut(idx) }.write((n % 10) as u8 + b'0');
+                    n /= 10;
+                    if n == 0 {
+                        break;
+                    }
+                }
+                unsafe {
+                    str::from_utf8_unchecked(
+                        &*(buffer.get_unchecked(idx..) as *const [_] as *const _),
+                    )
+                }
+            }
+
+            let mut debug_msg = ArrayString::<{ PANIC_OCCURRED.len() + TRIMMED_MAX_LEN }>::new();
+            let mut buffer = [MaybeUninit::uninit(); 10];
+
+            let _ = debug_msg.try_push_str(PANIC_OCCURRED);
+            let _ = debug_msg.try_push_str("'");
+            let _ = write!(&mut debug_msg, "{message}");
+            let _ = debug_msg.try_push_str("', ");
+            let _ = debug_msg.try_push_str(location.file());
+            let _ = debug_msg.try_push_str(":");
+            let _ = debug_msg.try_push_str(itoa_u32(&mut buffer, location.line()));
+            let _ = debug_msg.try_push_str(":");
+            let _ = debug_msg.try_push_str(itoa_u32(&mut buffer, location.column()));
+
+            let _ = ext::debug(&debug_msg);
+
+            let msg = unsafe { debug_msg.get_unchecked(PANIC_OCCURRED.len()..) };
+            ext::panic(&msg)
+        }
+
+        /// Panic handler for stable Rust <1.73.
+        #[rustversion::before(1.73)]
+        #[cfg(feature = "debug")]
+        #[cfg(not(feature = "panic-messages"))]
+        #[panic_handler]
+        pub fn panic(panic_info: &PanicInfo) -> ! {
+            use crate::prelude::fmt::Write;
+            use arrayvec::ArrayString;
+
+            static_assertions::const_assert!(PANICKED_AT.len() == (PANIC_OCCURRED.len() - 4));
+
+            let mut debug_msg = ArrayString::<{ PANIC_OCCURRED.len() + TRIMMED_MAX_LEN }>::new();
+
+            let _ = debug_msg.try_push_str(&PANIC_OCCURRED[..4]);
+            let _ = write!(&mut debug_msg, "{panic_info}");
+
+            let src = (&PANIC_OCCURRED[4..]).as_bytes();
+            let dest = unsafe {
+                debug_msg
+                    .as_bytes_mut()
+                    .get_unchecked_mut(4..PANIC_OCCURRED.len())
+            };
+            dest.copy_from_slice(src);
```

Comment:
Can be simplified
```suggestion
            debug_msg
                .as_bytes_mut()
                .get_unchecked_mut(4..PANIC_OCCURRED.len())
                .copy_from_slice(&PANIC_OCCURRED[4..])
```

Reply:
also `.as_bytes_mut()` and `get_unchecked_mut()` are `unsafe fn`
`.as_bytes()` is still needed
```
  173 |                 .copy_from_slice(&PANIC_OCCURRED[4..]);
      |                  --------------- ^^^^^^^^^^^^^^^^^^^^ expected `&[u8]`, found `&str`
```

Reply:
b23ab7b7ae9cf6f4d3eb68e6215a88f72a526243